api/lib/opentelemetry/trace/sampling_hint.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Trace
+    # Hints to influence sampling decisions. The default option for span
+    # creation is to not provide any suggestion.
+    module SamplingHint
+      # Suggest to not record events and not propagate.
+      NOT_RECORD = :__not_record__
```

Comment:
Is there no better idiom for an enum here?  I feel like I would just used the :blah everywhere.

Reply:
Google for `enums in ruby` for many many answers. Seems like everyone's favourite project is implementing enums in Ruby. We could pick one and pull it in, but I feel like it isn't worth the extra dependency.